client/src/config/settings.ts

```diff
@@ -18,6 +18,12 @@ export interface IPreferencesContext {
   wordWrap: boolean
 }
 
+export type Entries<T> = {
+    [K in keyof T]-?: [K, T[K]];
+}[keyof T][];
```

Comment:
I don't fully understand this magic, but maybe you could move it somewhere, e.g. under `utils/`? can also be a new file under `utils/`.

I've tried to keep `settings.ts` as clean as possible so it's easier to add more settings in the future.